src/main/scala/org/bytedeco/sbt/javacpp/JavaCppPlugin.scala

```diff
@@ -0,0 +1,140 @@
+package org.bytedeco.sbt.javacpp
+
+import java.net.URLClassLoader
+import java.util.Properties
+
+import sbt.Keys._
+import sbt._
+
+import scala.language.{ postfixOps, reflectiveCalls }
+import scala.util.Try
+
+object JavaCppPlugin extends AutoPlugin {
+
+  override def projectSettings: Seq[Setting[_]] = {
+    import autoImport._
+    Seq(
+      autoCompilerPlugins := true,
+      javaCppClasses := Seq.empty,
+      javaCppCustomizer := identity,
+      javaCppPlatform := Platform.current,
+      javaCppPresetLibs := Seq.empty,
+      javaCppVersion := Versions.javaCppVersion,
+      libraryDependencies += {
+        "org.bytedeco" % "javacpp" % javaCppVersion.value jar
+      },
+      javaCppPresetDependencies,
+      javaCppBuild := javaCpp.value,
+      products in Compile := (products in Compile).dependsOn(javaCppBuild).value)
+  }
+
+  object Versions {
+    val javaCppVersion = "1.4.3"
+  }
+
+  object autoImport {
+    type JavaCppBuilder = {
```

Comment:
If the real `org.bytedeco.javacpp.tools.Builder` is updated/changed (e.g. when the user configures a newer version), would this still work?

Reply:
It should work as long as the corresponding methods that are actually invoked are not missing in the newer version of `org.bytedeco.javacpp.tools.Builder`.